ethcore/sync/src/chain/handler.rs

```diff
@@ -87,9 +87,26 @@ impl SyncHandler {
 				Ok(())
 			}
 		};
-		result.unwrap_or_else(|e| {
-			debug!(target:"sync", "{} -> Malformed packet {} : {}", peer, packet_id, e);
-		})
+
+		match result {
+			Err(DownloaderImportError::Invalid) => {
+				debug!(target:"sync", "{} -> Malformed packet {}", peer, packet_id);
```

Comment:
Most of times `DownloaderImportError::Invalid` doesn't means malformed rlp, but some other issue which is separately traced (like `Ignored ancient new block hash`, `network id mismatch`, ...). Probably add separate variant for the decoder error (and why not to keep the decoder error like `DownloaderImportError::Decoder(DecoderError)` - isn't it helpful)?

Reply:
I don't think that `DecoderError` is useful here. There is nothing nothing useful we can do with this information.

And maybe, I'll just improve the error message, since the meaning of `Invalid` is broader than 'Malformed packet'. Would this be good enough?

Reply:
Sure :)